app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -421,6 +424,155 @@ private void loadIcal(BufferedReader in) {
+				JSONArray linklist = conference.getJSONArray("social_links");
+				for (i = 0; i < linklist.length(); ++i) {
+					JSONObject link = linklist.getJSONObject(i);
+					Schedule.Link slink = new Link(link.getString("link"), link.getString("name"));
+					slink.setType(link.optString("type", null));
+					links.addLast(slink);
+				}
+
+				//Adding website separately because it is not in social_links
+				if (conference.has("event_url")) {
+					Schedule.Link slink = new Link(conference.getString("event_url"), "Website");
+					links.addLast(slink);
+				}
+
+
+				//Adding website separately because it is not in social_links
+				if (conference.has("ticket_url")) {
+					Schedule.Link slink = new Link(conference.getString("ticket_url"), "Ticket URL");
+					links.addLast(slink);
+				}
+
+			}
+
+			//Getting microlocations to add latitude and longitude
+			JSONArray microlocations = conference.getJSONArray("microlocations");
+
+			//An array of pairs to have locations as key and latitude and longitude as value
+			HashMap<String, String> locs = new HashMap<>();
+			for (i = 0; i < microlocations.length(); i++) {
+
+				JSONObject room = microlocations.getJSONObject(i);
+				locs.put(room.getString("name"), room.getString("longitude") + "," + room.getString("latitude"));
+			}
+
+
+			//The sessions are contained in the array present in an object
+			JSONArray events = conference.getJSONArray("sessions");
+
+			for (i = 0; i < events.length(); i++) {
+
+				JSONObject event = events.getJSONObject(i);
+				String uid = event.getString("id");
+				String title = event.getString("title");
+
+                /*Our date format is different and I changed getTimeInMillis() a bit to ignore "+08"
+                in second part to avoid error in integer parsing*/
+				String startTimeS = event.getString("start_time");
+				String endTimeS = event.getString("end_time");
+				Date startTime, endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item = new Schedule.Item(uid, title, startTime, endTime);
+				item.setDescription(event.getString("long_abstract"));
+
+				if (event.getString("signup_url") != "null") {
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+
+				JSONObject microlocation = event.getJSONObject("microlocation");
+				String location = microlocation.getString("name");
+
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+
+				//Getting value (latitude and longitude) from the map by key (name)
+				String locString = locs.get(line.getTitle());
+				String latitude = locString.substring(0, locString.indexOf(','));
+				String longitude = locString.substring(locString.indexOf(',') + 1);
+
+				//Adding location details here
+				String latlon = null;
+				try {
+					latlon = ("geo:0,0?q=" + longitude + "," +
```

Comment:
Ah, shame, did I put URL-encoding of that in a not very reusable place? I'll refactor that later I guess.